src/Cassandra.IntegrationTests/Core/ClientTimeoutTests.cs

```diff
@@ -178,5 +178,33 @@ public void Should_Throw_NoHostAvailableException_When_All_Hosts_Down()
                 testCluster.ResumeNode(2);
             }
         }
+
+        /// Tests a TimeoutException is raised when all hosts down with default read timeout
+        ///
+        /// Should_Abort_After_Default_ReadTimeoutMillis tests that the driver throws a TimeoutException and aborts the 
+        /// query execution when all nodes in the cluster is down, given a the default ReadTimeoutMillis of 12 seconds.
+        ///
+        /// @since 3.0.0
+        /// @jira_ticket CSHARP-362
+        /// @expected_result A TimeoutException should be raised after 12 seconds.
+        ///
+        /// @test_category connection:timeout
+        [Test]
+        public void Should_Abort_After_Default_ReadTimeoutMillis()
+        {
+            var testCluster = TestClusterManager.GetNonShareableTestCluster(2, 1, true, false);
```

Comment:
Nit: use `TestClusterManager.CreateNew()` to create test clusters, the other methods like `GetNonShareableTestCluster()` are deprecated.